org.eclipse.scout.rt.server/src/main/java/org/eclipse/scout/rt/server/AbstractServerSession.java

```diff
@@ -272,9 +290,12 @@ protected void cancelRunningJobs() {
 
   protected void inactivateSession() {
     try {
-      BEANS
-          .optional(ILogoutService.class)
-          .ifPresent(ILogoutService::logout);
+      boolean backgroundSession = TypeCastUtility.castValue(getData(BACKGROUND_SESSION), boolean.class);
+      if (!backgroundSession) {
+        BEANS
```

Comment:
Suggestion: Since background sessions are not used in Scout yet, just move this code into a separate method and let the concrete application handling this case as desired:
```
  protected void inactivateSession() {
    try {
      logoutSession();
    }
    finally {
      m_active = false;
      m_stopping = false;
      fireSessionChangedEvent(new SessionEvent(this, SessionEvent.TYPE_STOPPED));
      m_sessionMetrics.sessionDestroyed(SESSION_TYPE);
      LOG.info("Server session stopped [session={}, user={}]", this, getUserId());
    }
  }

  protected void logoutSession() {
    BEANS
        .optional(ILogoutService.class)
        .ifPresent(ILogoutService::logout);
  }
```